src/IO/H5/VolumeData.cpp

```diff
@@ -348,6 +350,11 @@ void VolumeData::write_volume_data(
     h5::write_data(observation_group.id(), pole_connectivity,
                    {pole_connectivity.size()}, "pole_connectivity");
   }
+  // Write the serialized domain
+  if (serialized_domain.has_value()) {
+    h5::write_to_attribute(observation_group.id(), "domain",
```

Comment:
attributes have to be pretty small, I think under 64KB. Are we sure we won't reach that for larger domains? E.g. if the functions of time are written we will probably exceed that with the Shape map. And I think it's the _total_ size of all attributes together needs to be under 64KB. Why not just write this into a dataset instead?